content/en/profiler/enabling/java.md

```diff
@@ -94,6 +94,117 @@ java \
+
+For JMC users, the JFR CPU sample event `jdk.ExecutionSample` is replaced by the `datadog.ExecutionSample` event.
+
+#### Linux settings
+
+The CPU engine works on most systems, but if the value of `/proc/sys/kernel/perf_event_paranoid` is set to 3, the profiler can't use perf events to schedule CPU sampling. This results in degraded profile quality, falling back to using `itimer`. Set `/proc/sys/kernel/perf_event_paranoid` to 2 or lower with the following command:
+
+```
+sudo sh -c 'echo 2 >/proc/sys/kernel/perf_event_paranoid'
+```
+
+### Async Profiler Wallclock engine
+
+The wallclock profiling engine is useful for profiling latency and integrates tightly with APM tracing.
+The engine samples all threads, on- or off-cpu, with active tracing activity and can be used to diagnose trace or span latency.
+The engine is disabled by default, but can be enabled with:
+
+```
+# before 1.3.0 export DD_PROFILING_ASYNC_ENABLED=true
+export DD_PROFILING_ASYNC_WALL_ENABLED=true
+```
+
+or:
+
+```
+# before 1.3.0 -Ddd.profiling.async.enabled=true
+-Ddd.profiling.async.wall.enabled=true
+```
+
+For JMC users, the `datadog.MethodSample` event is emitted for wallclock samples.
+
+The wallclock engine does not depend on the `/proc/sys/kernel/perf_event_paranoid` setting.
+
+### Async Profiler Allocation Engine
+
+The allocation profiling engine can be enabled to contextualize allocation profiles, which supports allocation profiles filtered by endpoint.
+It is disabled by default but can be enabled with:
+
+```
+# before 1.3.0 export DD_PROFILING_ASYNC_ENABLED=true
+export DD_PROFILING_ASYNC_ALLOC_ENABLED=true
+```
+
+or:
+
+```
+# before 1.3.0 -Ddd.profiling.async.enabled=true
+-Ddd.profiling.async.alloc.enabled=true
+```
+
+For JMC users, the `datadog.ObjectAllocationInNewTLAB` and `datadog.ObjectAllocationOutsideTLAB` events replace the `jdk.ObjectAllocationInNewTLAB` and `jdk.ObjectAllocationOutsideTLAB` events.
+
+The allocation engine does not depend on the `/proc/sys/kernel/perf_event_paranoid` setting.
+
+### Collecting native stack traces
+
+With Async Profiler CPU or Wallclock engines enabled, native stack traces can be collected.
+Native stack traces include things like JVM internals, native libraries used by your application or the JVM, and syscalls.
+Native stack traces are not collected by default because most of the time they do not provide actionable insights and there is walking native stacks creates a risk to application stability.
+To enable native stack trace collection at your own risk, set:
+
+```
+# before 1.3.0 export DD_PROFILING_ASYNC_ENABLED=true
+export DD_PROFILING_ASYNC_CSTACK=dwarf
+```
+
+or:
+
+```
+# before 1.3.0 -Ddd.profiling.async.enabled=true
+-Ddd.profiling.async.cstack=dwarf
+```
+
+In many situations, this setting can be used safely, but there are no guarantees.
+Ensure that you test this setting in a non-production environment before using it in production.
+
+### Async Profiler Minimum JDK Versions
```

Comment:
Maybe less confusing to move (an adapted version of) this section to the "Requirements" section at the top.

This may be clear but just in case not- the ~architecture of this doc is supposed to be something like:
- ~quickstart at the top
- all the details for doing fancier things below
Hopefully many users can just read roughly the first screenful of instructions and be off to the races.